relay-server/src/envelope.rs

```diff
@@ -554,6 +554,17 @@ pub struct ItemHeaders {
     #[serde(default, skip_serializing_if = "is_false")]
     metrics_extracted: bool,
 
+    /// Whether or not a transaction has been extracted from a segment span.
+    #[serde(default, skip_serializing_if = "is_false")]
+    transaction_extracted: bool,
+
+    /// Whether or not spans and span metrics have been extracted from a transaction.
+    ///
+    /// This header is set to `true` after both span extraction and span metrics extraction,
+    /// and can be used to skip extraction.
+    #[serde(default, skip_serializing_if = "is_false")]
+    spans_extracted: bool,
```

Comment:
I stumbled over the usage of these fields a bit and I am wondering if it makes sense to change the behaviour a bit. Please bear with me and correct me if I misunderstood something.

If there is a transaction extracted from a span:
- The span gets `transaction_extracted = true`
- The new transaction gets `span_extracted = true` (that isn't actually true, we just need it so we skip span extraction, but this transaction never had a span extracted)

Would it make more sense to have the opposite behaviour where:
- The span gets nothing (except we need this info for some other reason later down the road)
- The transaction gets `extracted_from_span = True` (or `extracted = True / Span`)
- A span extracted from a transaction would get `extracted_from_transaction = True` (or `extracted = True / Transaction`)

We can use an `enum` for `extracted` which means we should be nicely future proof and it is quite readable (imo) and we indicate where something came instead of the opposite.

Reply:
We discussed this offline and came to the conclusion we should keep the current naming.

> I'm trying your suggestion [here](https://github.com/getsentry/relay/pull/3375/files#r1560777429) and it works, but I have two problems:
I still want to mark spans as transaction_extracted even if it happens only in processing relays, just to be robust against refactors
An extracted_from_span header is more truthful, but a spans_extracted header seems more robust against refactors (same argument as above, if we move any extraction code to edge relays it might be nice to prevent double extractions)
I.e. I believe we should have had a spans_extracted flag from the start. In theory we could have both an extracted_from_span and a spans_extracted header, but the mere amount of flags would make me dizzy

The alternative of using `skip_*_extraction` came up, but we may want to stick with the state rather then the behaviour so we can use that information (state) later on for more actions that just skipping extraction.